python/graphstorm/gconstruct/transform.py

```diff
@@ -507,7 +607,17 @@ def parse_label_ops(confs, is_node):
     label_conf = confs[0]
     assert 'task_type' in label_conf, "'task_type' must be defined in the label field."
     task_type = label_conf['task_type']
-    # By default, we use all labels for training.
+    if 'custom_split_filenames' in label_conf:
+        custom_split = label_conf['custom_split_filenames']
+        assert isinstance(custom_split, dict), \
+                "Custom data split needs to provide train/val/test index."
+        train_idx = read_index_json(custom_split['train']) if 'train' in custom_split else None
```

Comment:
Can the index file be a parquet or hdf5?

Reply:
i'm following the format of Neptune ML right now. it can be extended to use any format. maybe we can support numpy. i'm not sure if we should support parquet or hdf5.